client/state/initial-state.js

```diff
@@ -94,8 +104,10 @@ export function persistOnChange( reduxStore, serializeState = serialize ) {
 }
 
 export default function createReduxStoreFromPersistedInitialState( reduxStoreReady ) {
-	if ( config.isEnabled( 'persist-redux' ) && ! isSupportUserSession() ) {
-		localforage.getItem( 'redux-state' )
+	if ( config.isEnabled( 'persist-redux' ) && isLoggedIn() && ! isSupportUserSession() ) {
+		// clear storage any storage that used the old-style key
+		localforage.removeItem( 'redux-state' );
```

Comment:
Hmm, perhaps use https://localforage.github.io/localForage/#data-api-clear ? The only other thing we're storing with localForage at the moment is the sync-handler which basically just caches some post list data. I can't think of a good reason for that to stick around when people log out.